Zend/tests/match/020.phpt

```diff
@@ -0,0 +1,39 @@
+--TEST--
+Test breaking out of match arm with continue
+--FILE--
+<?php
+
+foreach (range(0, 9) as $i) {
+    match ($i) {
+        default => {
+            echo "$i 1\n";
+            continue;
+            echo "$i 2\n";
+        },
+    }
+
+    echo "$i 3\n";
+}
+
+--EXPECTF--
+Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in %s on line %d
```

Comment:
```suggestion
Warning: "continue" targeting switch or match is equivalent to "break". Did you mean to use "continue 2"? in %s on line %d
```

Reply:
Another alternative is forbidding it entirely for `match` - For switch, there's pre-existing code, so it was introduced as a warning, but there's no need to be forced to support this for the brand new `match`.

Something along these lines:
`E_COMPILE_ERROR Error: "continue" targeting match is forbidden, it would be equivalent to "break". Did you mean to use "continue 2" or "break"? in %s on line %d`

Reply:
`continue` should actually mean continue any loops outside the match expression. We should probably fix this for both at some point. Not sure if this is possible for the switch as it was only a warning until now. Not sure what I think about fixing it only for match.

Reply:
I'd be worried about the language inconsistency and end user confusion about

- `continue` really meaning `continue 2`
- `break N;` going to a different level than `continue N;`, in loops/switches containing match expressions
- `continue N` and `break N` in `for(;;) {match() {... match() { X => { break 2; }}}`

Reply:
> We should probably fix this for both at some point. Not sure if this is possible for the switch as it was only a warning until now. Not sure what I think about fixing it only for match.

That'd definitely be a backwards incompatibility break I'd strongly want to avoid - the same code `continue N;` would unexpectedly stop working in php 8, and my comment about `break N;` and `continue N;` referring to different scopes would also apply.

Reply:
Yeah I suppose that's true.